### PR TITLE
MAD-X Parser: Thick Solenoid

### PR DESCRIPTION
**Pull request by ax3l**
Add support to read thick solenoids from MAD-X files.

- [x] add test

### CODE REVIEW

**Review by n01r — APPROVED** [review]

File: examples/solenoid/run_solenoid_madx.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+sim.space_charge = False
+# sim.diagnostics = False  # benchmarking
+sim.slice_step_diagnostics = True
+
+# domain decomposition & space charge mesh
+sim.init_grids()
+
+# load a 250 MeV proton beam with an initial
+# normalized transverse rms emittance of 1 um
+energy_MeV = 250.0  # reference energy
```
Comment (ax3l) [inline examples/solenoid/run_solenoid_madx.py]: unused and loaded from `solenoid.madx`
```suggestion
```

File: examples/solenoid/run_solenoid.py [hunk 2]
```diff
@@ -20,7 +20,7 @@
 # domain decomposition & space charge mesh
 sim.init_grids()
 
-# load a 2 GeV proton beam with an initial
+# load a 250 MeV proton beam with an initial
 # normalized transverse rms emittance of 1 um
```
Comment (cemitch99) [inline examples/solenoid/run_solenoid.py]: The documentation should read:

"load a 250 MeV proton beam with an initial horizontal rms emittance of 1 um and an initial vertical rms emittance of 2 um"

**Review by cemitch99 — APPROVED** [review]
Looks ok to me.  Note minor comment added on "run_solenoid.py".

File: examples/solenoid/run_solenoid.py (hunk 2 above)
Comment (ax3l) [reply examples/solenoid/run_solenoid.py]: Thanks a lot, fixed in both scripts.

**Pull request merged**